In [1]:
import os
os.chdir(os.getcwd() + "\..")

In [2]:
from src import damage

## Кирара

### Базовые статы + таланты

In [3]:
kirara_base_hp = 12180
kirara_base_atk = 223
kirara_hp_ascention = 0.2

def kirara_shield_absorbtion(hp):
    return 0.17 * hp + 1956

def kirara_max_shield_absorbtion(hp):
    return 0.272 * hp + 3133

def kirara_reinforced_packaging(hp):
    return kirara_shield_absorbtion(hp) * 0.2

kirara_ult_mv = 9.694 + 0.606 * 4

### Билд

Допустим, дабл ХП + 40% из подстатов. В шапке крит, что фав факинг прокал. Допустим она на дендро сете.

In [4]:
artifacts_hp_percent = 0.466 * 2 + 0.4
artifacts_flat_hp = 4000
# Реализуемо ок? ок
artifacts_atk_percent = 0.2
artifacts_flat_atk = 400
favge_base = 454

### Сколько же танчит ее щит?

In [5]:
kirara_total_hp = damage.total_hp(kirara_base_hp, artifacts_hp_percent + kirara_hp_ascention, artifacts_flat_hp)
kirara_total_hp

34839.76

Базовое ХП щита и максимальное

In [6]:
damage.shield_hp(kirara_shield_absorbtion(kirara_total_hp), 1, 1)

7878.7592

In [7]:
damage.shield_hp(kirara_max_shield_absorbtion(kirara_total_hp), 1, 1, 1)

12609.41472

Что если урон - дендро (что вполне реально в блум и барджион командах)

In [8]:
print(damage.shield_hp(kirara_shield_absorbtion(kirara_total_hp), 1, 2.5))
print(damage.shield_hp(kirara_max_shield_absorbtion(kirara_total_hp), 1, 2.5))

19696.898
31523.5368


Дополнительные щиты от ее А1. За одну ешку не смогут настакать максимум.

In [8]:
damage.shield_hp(kirara_reinforced_packaging(kirara_total_hp), 1, 1)

1575.7518400000001

### Сколько же наносит ее ульта?

In [9]:
kirara_total_atk = damage.total_atk(kirara_base_atk + favge_base, artifacts_atk_percent, artifacts_flat_atk)
kirara_a4_ult_buff = kirara_total_hp / 1000 * 0.003
effective_talent = damage.effective_talent(kirara_total_atk, cd=1, cr=0.5, crit_ratio=0.5, dmg_bonus=1 + 0.15 + kirara_a4_ult_buff)
talent_dmg = damage.talent_dmg(effective_talent, kirara_ult_mv)
outgoing_dmg = damage.outgoing_dmg_multiplicative(talent_dmg, character_level=90, enemy_level=100)
outgoing_dmg

21784.800739393642

In [13]:
kirara_a4_ult_buff

0.10451928000000002

10% BAAAATCHEEEST

## Дед

### Базовые статы + таланты

In [10]:
ded_base_hp = 14695
ded_base_atk = 251

def ded_shield_absorbtion(hp):
    return 0.218 * hp + 2506


Билд тот же самый, что и у Кирары (и тоже дендро сет, потому что дед может прокать дендро сет)

### Что по щиту? - самый прочный в игре

In [11]:
ded_total_hp = damage.total_hp(ded_base_hp, artifacts_hp_percent, artifacts_flat_hp)
ded_total_hp

38268.74

In [12]:
damage.shield_hp(ded_shield_absorbtion(ded_total_hp), 1, shield_strength=1 + 0.05*5, damage_absorbtion=1.5)

20341.097475

## И что в итоге:

Кирара:

- Макс прочность щита 12609 (также учитываем, что эти 12к тебе "даются" не сразу) (но от дендро урона прочность возрастает до 31к!!!)

- 100% аптайм щита

- Анимация прожатия щита почти моментальная

- Энергонезависимая

- Солидный нюк на ульте, который не жалко использовать

- Нюк на ульте накладывает ДЕНДРО

- Режет дендро резист на 30%

- **Может накладывать дендро**

- Дендро: не выключает Нилу, закрывает дендро резонанс

- Милилит почти не работает (там где он нужен, как правило ты не захочешь находиться в ешке)

- Кошкодевочка

Дед:

- Макс прочность щита 20341

- 100% аптайм щита

- Анимация прожатия щита долгая и неудобная

- Энергонезависимый

- Нюк на ульте, который ты не будешь использовать (петрификация бесполезная кста)

- Нюк на ульте ничего не накладывает

- Режет дендро резист на 30% + все резисты на 20%

- Ничего не накладывает, лол

- Гео, лол

- Милилит хорошо работает

- Какой-то старый дед